src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1838,6 +1839,69 @@ static SimpleType[][] GetObjectwith2DArrayOfSimpleType()
         simpleType2D[1][1] = new SimpleType() { P1 = "1 1 value", P2 = 4 };
         return simpleType2D;
     }
+
+    [Fact]
+    public static void XmlSchemaTest()
+    {
+        XmlSchemas schemas = new XmlSchemas();
+        XmlSchemaExporter exporter = new XmlSchemaExporter(schemas);
+        //Import the type as an XML mapping
+        XmlTypeMapping originalmapping = new XmlReflectionImporter().ImportTypeMapping(typeof(Dog));
+        //Export the XML mapping into schemas
+        exporter.ExportTypeMapping(originalmapping);
+        //Print out the schemas
+        XmlSchemaEnumerator schemaEnumerator = new XmlSchemaEnumerator(schemas);
+        MemoryStream ms = new MemoryStream();
+        string expected = "<?xml version=\"1.0\"?>\r\n<xs:schema elementFormDefault=\"qualified\" xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">\r\n  <xs:element name=\"Dog\" nillable=\"true\" type=\"Dog\" />\r\n  <xs:complexType name=\"Dog\">\r\n    <xs:complexContent mixed=\"false\">\r\n      <xs:extension base=\"Animal\">\r\n        <xs:sequence>\r\n          <xs:element minOccurs=\"1\" maxOccurs=\"1\" name=\"breed\" type=\"DogBreed\" />\r\n        </xs:sequence>\r\n      </xs:extension>\r\n    </xs:complexContent>\r\n  </xs:complexType>\r\n  <xs:complexType name=\"Animal\">\r\n    <xs:sequence>\r\n      <xs:element minOccurs=\"1\" maxOccurs=\"1\" name=\"age\" type=\"xs:int\" />\r\n      <xs:element minOccurs=\"0\" maxOccurs=\"1\" name=\"name\" type=\"xs:string\" />\r\n    </xs:sequence>\r\n  </xs:complexType>\r\n  <xs:simpleType name=\"DogBreed\">\r\n    <xs:restriction base=\"xs:string\">\r\n      <xs:enumeration value=\"GermanShepherd\" />\r\n      <xs:enumeration value=\"LabradorRetriever\" />\r\n    </xs:restriction>\r\n  </xs:simpleType>\r\n</xs:schema>";
+        while (schemaEnumerator.MoveNext())
```

Comment:
I wondered if we want a `while` loop here. We know we have only one schema. I think we'd better Assert that in this test.